playwright/src/fixtures/MHmodal.ts

```diff
@@ -23,6 +29,101 @@ class MHModal {
     // Wait for the highlights to be loaded in MH modal
     await this.page.waitForSelector('[class*=HighlightOuterWrapper]')
   }
+
+  async chapterDropdownCount() {
+    // Total number of checkboxes in the chapter dropdown
+    const Toc = new TOC(this.page)
+    const chapterDropdownCount = (await Toc.chapterCount()) + (await Toc.standalonePagesCount())
+    return chapterDropdownCount
+  }
+
+  async toggleChapterDropdown() {
+    // Open the chapter dropdown filter
+    await this.chapterDropdownLocator.click()
+  }
+
+  async toggleColorDropdown() {
+    // Open the color dropdown filter
+    await this.colorDropdownLocator.click()
+  }
+
+  async colorNumber(color: string) {
+    // Assign number to each color in the color dropdown
+    if (color === 'yellow') {
+      return '0'
+    } else if (color === 'green') {
+      return '1'
+    } else if (color === 'blue') {
+      return '2'
+    } else if (color === 'purple') {
+      return '3'
+    } else if (color === 'pink') {
+      return '4'
+    }
```

Comment:
This should be replaced by a switch statement
```
switch (color) {
  case 'yellow':
    return '0'
  ...
}
```
or a Class definition (_if it can/will be used in other parts of the code_)